### PR TITLE
chore: update getPage to use apollo

### LINKED ISSUE #198: Update `getPage` from api-client
Currently, the {{getPage }}from api-client uses a shopify-buy library to interact with shopify API.

We need to migrate and drop the dependency on shopify-buy and start using Apollo GraphQL.

**Disclaimer: Remember to add unit tests and proper documentation.**

┆Issue is synchronized with this [Jira Zadanie](https://vsf.atlassian.net/browse/SH-52) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by jeffpdotone**
Closes #198

## Description
This is an update to use the graphql query page through the apollo client

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/apollo/src/library.ts [hunk 1]
```diff
@@ -31,8 +31,16 @@ export * from './api'
...
 
+type ExtendQueryResult = { query: any; variables: any }
 
-export type ShopifyApolloContext = IntegrationContext<ShopifyApolloClient, ShopifyApolloSettings, ShopifyApolloAPIs>
+type ExtendQueryContext = {
+  extendQuery: (
+    customQuery: CustomQuery,
+    query: Record<string, ExtendQueryResult>
+  ) => Record<string, ExtendQueryResult>
+}
```
Comment (Mateuszpietrusinski) [inline packages/apollo/src/library.ts:41]: (https://blog.logrocket.com/types-vs-interfaces-in-typescript/)
Interfaces are basically a way to describe data shapes, for example, an object.

Type is a definition of a type of data, for example, a union, primitive, intersection, tuple, or any other type.

File: packages/composables/src/useContent/index.ts [hunk 2]
```diff
@@ -12,7 +12,8 @@ const params: UseContentFactoryParams<ContentType, ContentPosition> = {
       if (params.ContentType === 'updatePreferences') {
         return await context.$shopify.api.updateNewsLetterPreferences(searchParams);
       } else if (params.ContentType === 'page') {
-        return await context.$shopify.api.getPages(searchParams);
+        const response = await context.$shopify.api.getPage(params)
+        return response.data.page;
```
Comment (Mateuszpietrusinski) [inline packages/composables/src/useContent/index.ts:16]: Optional chaining in case of errors
```suggestion
        return response?.data?.page;
```

**Review by Mateuszpietrusinski — CHANGES_REQUESTED** [review]
Requires adjustments

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# top-most EditorConfig file
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = false
+insert_final_newline = false
```
[the remark this answers is not in the document]
Comment (jeffpdotone) [reply .editorconfig:12]: so far it's aligned with prettier & eslint

File: packages/apollo/__tests__/searchProduct.spec.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      first: 10
+    }
+
+    mockExtendQuery.mockImplementationOnce(() => ({ products: { query: 'test-search-query', variables: expectedVariables } }))
+
+    await searchProduct(mockContext, params, customQuery)
+
+    expect(mockExtendQuery).toHaveBeenCalledWith(customQuery, {
+      products: {
+        query: expect.any(Object),
```
Comment (Mateuszpietrusinski) [inline packages/apollo/__tests__/searchProduct.spec.ts:38]: In long term the query should be provided here

File: packages/apollo/src/api/getPage.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      }
+    }
+  )
+
+  const response = await context.client.apolloClient.query<QueryRoot, QueryRootPageArgs>({
+    query: page.query,
+    variables: page.variables
+  })
+
+  return response ?? null
```
Comment (Mateuszpietrusinski) [inline packages/apollo/src/api/getPage.ts:46]: allocation is redundant 
```suggestion
  return await context.client.apolloClient.query<QueryRoot, QueryRootPageArgs>({
    query: page.query,
    variables: page.variables
  }) ?? null
```

File: packages/apollo/src/types/ExtendQueryContext.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { CustomQuery } from "@vue-storefront/core";
+
+interface ExtendQueryParams { query: any; variables: any }
+
+export default interface ExtendQueryContext {
```
Comment (Mateuszpietrusinski) [inline packages/apollo/src/types/ExtendQueryContext.ts:5]: Use without default
```suggestion
export interface ExtendQueryContext {
```

File: packages/apollo/src/library.ts [hunk 7]
```diff
@@ -1,16 +1,10 @@
...
-    storefrontAccessToken: string
-    cookies?: string
-  },
-  currency: string
-  country: string
-}
+import type { ShopifyApolloAPIs } from './api'
+import ShopifyApolloSettings from "./types/ShopifyApolloSettings"
+import ShopifyApolloClient from "./types/ShopifyApolloClient"
+import ExtendQueryContext from "./types/ExtendQueryContext"
```
Comment (Mateuszpietrusinski) [inline packages/apollo/src/library.ts:7]: those interfaces can be imported from destructuration instead as default

File: packages/apollo/src/types/ShopifyApolloClient.ts [hunk 8]
```diff
@@ -0,0 +1,3 @@
+import type { createShopifyApollo } from "../library";
+
+export default interface ShopifyApolloClient { apolloClient: ReturnType<typeof createShopifyApollo> }
```
Comment (Mateuszpietrusinski) [inline packages/apollo/src/types/ShopifyApolloClient.ts:3]: same as above

File: packages/apollo/src/types/ShopifyApolloSettings.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+export default interface ShopifyApolloSettings {
```
Comment (Mateuszpietrusinski) [inline packages/apollo/src/types/ShopifyApolloSettings.ts:1]: same as above

File: packages/apollo/src/types/getPage/GetPageParams.ts [hunk 10]
```diff
@@ -0,0 +1,4 @@
+export default interface GetPageParams {
```
Comment (Mateuszpietrusinski) [inline packages/apollo/src/types/getPage/GetPageParams.ts:1]: same as above

**Review by Mateuszpietrusinski — CHANGES_REQUESTED** [review]
Some warnings are in the pr, check them. One question, why do you prefer to use default export instead of direct export/import? + one improvment

File: packages/apollo/__tests__/searchProduct.spec.ts (hunk 4 above)
Comment (jeffpdotone) [reply packages/apollo/__tests__/searchProduct.spec.ts:38]: Will provide this in the next unit tests

File: packages/api-client/src/api/customerMutations/buildQueries.ts [hunk 11]
```diff
@@ -69,8 +69,7 @@ const ordersQuery: (pages: number, token: string, context) => any = (pages, toke
   });
 };
 
-const addressesQuery: (pages: number, token: string, context) => any = (pages, token, context): any => {
-
+const addressesQuery: (pages: number, token: string, context) => any = (_pages, token, context): any => {
```
Comment (Mateuszpietrusinski) [inline packages/api-client/src/api/customerMutations/buildQueries.ts:72]: Can we add the returning type here?

File: packages/api-client/src/api/fetchOrders/index.ts [hunk 12]
```diff
@@ -1,10 +1,4 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
-import { CustomQuery } from '@vue-storefront/core';
-import { ordersQuery as query } from './../customerMutations/buildQueries';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-export default async function fetchOrders(context, params, customQuery?: CustomQuery) {
+export default function fetchOrders(context, params) {
```
Comment (Mateuszpietrusinski) [inline packages/api-client/src/api/fetchOrders/index.ts:1]: Just being curious are we now passing customQuery in params?

**Review by Mateuszpietrusinski — APPROVED** [review]
small questions

File: packages/api-client/src/api/fetchOrders/index.ts (hunk 12 above)
Comment (jeffpdotone) [reply packages/api-client/src/api/fetchOrders/index.ts:1]: This is are from old fetchOrders and doesn't implement yet the customQuery.

**Pull request merged**